cmd/internal/kube/secret.go

```diff
@@ -0,0 +1,148 @@
+package kube
+
+import (
+	"bytes"
+	"encoding/base64"
+	"os"
+	"path/filepath"
+	"text/template"
+
+	"github.com/akitasoftware/akita-cli/telemetry"
+
+	"github.com/akitasoftware/akita-cli/cmd/internal/cmderr"
+	"github.com/akitasoftware/akita-cli/printer"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	outputFlag    string
+	namespaceFlag string
+	// Store a parsed representation of /template/akita-secret.tmpl
+	secretTemplate *template.Template
+)
+
+var secretCmd = &cobra.Command{
+	Use:   "secret",
+	Short: "Generate a Kubernetes secret containing the Akita credentials",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		key, secret, err := cmderr.RequireAPICredentials("Akita API key is required for Kubernetes Secret generation")
+		if err != nil {
+			return err
+		}
+
+		output, err := handleSecretGeneration(namespaceFlag, key, secret, outputFlag)
+		if err != nil {
+			return err
+		}
+
+		// Output the generated secret to the console
+		printer.RawOutput(output)
```

Comment:
Sorry, I think I was unclear.  The two uses cases are:

1. Apply directly 
akita kube secret | kubectl apply -f -

2. Apply via a file (convenience, they could always pipe to a file.)
akita kube secret -f mysecret.yaml
kubectl apply -f mysecret.yaml

In case #2 we should not print to standard output as well, it should be one or the other.  We can support this in a few different ways, I don't much care whether (a) standard out is the default if `-f` not specified, or (b) `-f -` writes to standard output..

I kind of think like in case #1 we should not write the file that was not asked for as well.

Come talk to me if what should happen in these two cases are still unclear.